src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -1873,4 +1881,104 @@ private boolean expectedFunction(@Nullable PsiElement resolve) {
+
+      int position = args.getExpressionList().indexOf(expr);
+      if (position == -1) return false;
+
+      ErlangFunctionCallExpression call = ObjectUtils.tryCast(args.getParent(), ErlangFunctionCallExpression.class);
+      ErlangFunction function = resolveToFunction(call);
+      if (function == null) return false;
+
+      for (ErlangFunctionClause clause : function.getFunctionClauseList()) {
+        if (elementSuits(clause, position)) return true;
+      }
+      return false;
+    }
+
+    private boolean elementSuits(@NotNull ErlangFunctionClause clause, int position) {
+      List<ErlangArgumentDefinition> definitions = clause.getArgumentDefinitionList().getArgumentDefinitionList();
+      ErlangArgumentDefinition definition = definitions.size() > position ? definitions.get(position) : null;
+      ErlangMaxExpression maxExpression = PsiTreeUtil.getChildOfType(definition, ErlangMaxExpression.class);
+      ErlangQVar var = PsiTreeUtil.getChildOfType(maxExpression, ErlangQVar.class);
+      if (var != null && NAMES.contains(var.getName())) {
+        return true;
+      }
+      return false;
+    }
+  }
+
+  public static class ErlangSpecParameter<T extends PsiElement> extends PatternCondition<T> {
+    public ErlangSpecParameter() {
+      super("SpecParameter");
+    }
+
+    @Override
+    public boolean accepts(@NotNull T element, ProcessingContext context) {
+      ErlangQAtom atom = element instanceof ErlangQAtom ? (ErlangQAtom) element : null;
+      ErlangMaxExpression expr = ObjectUtils.tryCast(atom != null ? atom.getParent() : null, ErlangMaxExpression.class);
+      ErlangArgumentList args = ObjectUtils.tryCast(expr != null ? expr.getParent() : null, ErlangArgumentList.class);
+      ErlangSpecification specification = getSpecification(args);
+      if (atom == null || expr == null || args == null || specification == null) return false;
+
+      int position = args.getExpressionList().indexOf(expr);
+      if (position == -1) return false;
+
+      ErlangFunTypeSigs signature = specification.getSignature();
+      if (signature == null) return false;
+
+      for (ErlangTypeSig typeSig : signature.getTypeSigList()) {
+        List<ErlangType> types = typeSig.getFunType().getFunTypeArguments().getTypeList();
+        if (types.size() <= position) continue;
+
+        ErlangQVar typeVar = types.get(position).getQVar();
+        if (argumentIsModule(typeVar, typeSig)) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    private static boolean argumentIsModule(@Nullable ErlangQVar typeVar, @NotNull ErlangTypeSig typeSig) {
+      String typeVarName = typeVar != null ? typeVar.getName() : null;
+      ErlangTypeSigGuard sigGuard = typeSig.getTypeSigGuard();
+      if (sigGuard == null || typeVarName == null) return false;
+
+      for (ErlangTypeGuard guard : sigGuard.getTypeGuardList()) {
+        ErlangTopType topType = PsiTreeUtil.getChildOfType(guard, ErlangTopType.class);
```

Comment:
Again, avoid using PsiTreeUtil.getChildOfType - use guard.getTypeList() instead. Not only you'll re-use existing implementation of searching for types, but you'll also write correct code as opposed to what's written here - you only consider first type in a list.